### PR TITLE
(chores) camel-base-engine: cleanup duplicated code

### PR DESCRIPTION
**Pull request by orpiske**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/CamelInternalProcessor.java [hunk 1]
```diff
@@ -813,10 +801,10 @@ public void after(Exchange exchange, UnitOfWork uow) throws Exception {
                 UnitOfWorkHelper.doneUow(uow, exchange);
             }
 
-            // after UoW is done lets pop the route context which must be done on every existing UoW
             if (route != null && existing != null) {
                 existing.popRoute();
             }
+            // after UoW is done lets pop the route context which must be done on every existing UoW
```
Comment (davsclaus) [inline core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/CamelInternalProcessor.java:807]: this comment better belongs where it was before

**Review by davsclaus — COMMENTED** [review]
I guess there is no noticable performance hit with the extra method call? The method is too big for the JVM to inlining

**Comment by orpiske** [comment]
> I guess there is no noticable performance hit with the extra method call? The method is too big for the JVM to inlining


Yeah, this one is probably too big for the inlining with the default JVM settings. The greatest benefit of consolidating this code is having nicer stack frames with the async profiler (and that makes it easier to rule out the parts that are performing nicely).

**Review by davsclaus — APPROVED** [review]

File: core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/CamelInternalProcessor.java (hunk 1 above)
Comment (orpiske) [reply core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/CamelInternalProcessor.java:807]: Oh, alright. I'll merge and then move it back in place!

**Pull request merged**